pkg/compose/convergence.go

```diff
@@ -280,6 +294,9 @@ func (s *composeService) waitDependencies(ctx context.Context, project *types.Pr
 						return err
 					}
 					if healthy {
+						for _, container := range containers {
```

Comment:
To avoid such loop being repeated many times, I suggest to introduce `w.ContainerEvent(type, containers...)`

Reply:
Introducing a dependency on `moby.Container` in the `progress` package seemed a bit weird, so I split this into `progress.Writer.Events` and a `containerEvents` helper function in this file. WDYT? You should also feel free to just take over the PR, if that'd be easier!